Plugin/Model/Order/ItemRepositoryPlugin.php

```diff
@@ -0,0 +1,142 @@
+     * @param GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode
+     * @param SourceSelectionServiceInterface $sourceSelectionService
+     */
+    public function __construct(
+        OrderItemExtensionFactory $orderItemExtensionFactory,
+        GetSourceSelectionResultFromOrder $sourceSelectionResult,
+        GetInventoryRequestFromOrder $getInventoryRequestFromOrder,
+        GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode,
+        SourceSelectionServiceInterface $sourceSelectionService
+    ) {
+        $this->orderItemExtensionFactory = $orderItemExtensionFactory;
+        $this->sourceSelectionResult = $sourceSelectionResult;
+        $this->getInventoryRequestFromOrder = $getInventoryRequestFromOrder;
+        $this->getDefaultSourceSelectionAlgorithmCode = $getDefaultSourceSelectionAlgorithmCode;
+        $this->sourceSelectionService = $sourceSelectionService;
+    }
+
+    /**
+     * Set additional attributes when requesting a single order item
+     *
+     * @param ItemRepository  $subject
+     * @param OrderItemInterface|Item $result
+     * @return OrderItemInterface|Item
+     */
+    public function afterGet(ItemRepository $subject, OrderItemInterface $result)
+    {
+        return $this->applyExtensionAttributesToOrderItem($result, $result);
+    }
+
+    /**
+     * Set additional attributes when requesting a list of order items
+     *
+     * @param ItemRepository $subject
+     * @param OrderItemSearchResultInterface $result
+     * @return OrderItemSearchResultInterface
+     */
+    public function afterGetList(ItemRepository $subject, OrderItemSearchResultInterface $result)
+    {
+        foreach ($result->getItems() as $item) {
+            $this->applyExtensionAttributesToOrderItem($item, $result->getItems());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Apply extension attributes to order item
+     *
+     * @param OrderItemInterface $orderItem
+     * @param array $allItems
+     * @return OrderItemInterface
+     */
+    private function applyExtensionAttributesToOrderItem(OrderItemInterface $orderItem, array $allItems) : OrderItemInterface
+    {
+        if (!$extensionAttributes = $orderItem->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderItemExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSourceCode(
+            $this->getCustomAttributeValue($orderItem->getOrder(), $allItems)
+        );
+
+        $orderItem->setExtensionAttributes($extensionAttributes);
+
+        return $orderItem;
+    }
+
+    // @TODO what happens if there's more than one source?
+    /**
+     * @param Order $order
+     * @param array $allItems
+     * @return string|null
+     */
+    private function getCustomAttributeValue(Order $order, array $allItems)
+    {
+        $inventoryRequest = $this->getInventoryRequestFromOrder->execute(
```

Comment:
Is this fairly performant? Maybe we should be looking into caching the result for an order? Especially when iterating the order items (and essentially using the same data set)